cmd/download.go

```diff
@@ -0,0 +1,250 @@
+	}
+
+	builder.WriteString(fmt.Sprintf("Duration	[total]			%s\n", ds.endTime.Sub(ds.startTime).Round(time.Second)))
+
+	if len(ds.requestDurations) > 0 {
+		p := util.CalculateDurationStatistics(ds.requestDurations)
+		builder.WriteString(fmt.Sprintf("Requ. Latencies	[mean, 50, 95, 99, max]	%s, %s, %s, %s, %s\n", p.Mean, p.Q50, p.Q95, p.Q99, p.Max))
+	}
+
+	if len(ds.processingDurations) > 0 {
+		p := util.CalculateDurationStatistics(ds.processingDurations)
+		builder.WriteString(fmt.Sprintf("Proc. Latencies	[mean, 50, 95, 99, max]	%s, %s, %s, %s, %s\n", p.Mean, p.Q50, p.Q95, p.Q99, p.Max))
+	}
+
+	totalRequests := len(ds.requestDurations)
+	builder.WriteString(fmt.Sprintf("Bytes In	[total, mean]		%s, %s\n", util.FmtBytesHumanReadable(float32(ds.totalBytesIn)), util.FmtBytesHumanReadable(float32(ds.totalBytesIn)/float32(totalRequests))))
+
+	return builder.String()
+}
+
+var downloadCmd = &cobra.Command{
+	Use:   "download",
+	Short: "Download FHIR resources into an NDJSON file",
+	Long: `Downloads FHIR resources and puts them into an NDJSON file.
+	
+Potential FHIR resources that will be downloaded can be limited by a mandatory -t/--type flag
+and an optional -q/--query flag. The query flag has to be a valid FHIR search query.
+
+Downloaded resources will be stored within a file denoted by the -o/--output-file flag.
+
+Example:
+	
+	blazectl download -t List
+	blazectl download -t List -q "status=current"`,
+	Run: func(cmd *cobra.Command, args []string) {
+		sink := createOutputFileOrDie(outputFile)
+		defer sink.Close()
+
+		stats, err := downloadResources(client, resourceType, fhirSearchQuery, sink)
+		if err != nil {
+			fmt.Printf("Download failed: %v\n", err)
+			os.Exit(1)
+		}
+		fmt.Println(stats.String())
+	},
+}
+
+func downloadResources(client *fhir.Client, resourceType string, fhirSearchQuery string, sink io.Writer) (*downloadStats, error) {
+	var stats downloadStats
+	stats.startTime = time.Now()
+
+	query, err := url.ParseQuery(fhirSearchQuery)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse the FHIR search query: %v\n", err)
+	}
+
+	var request *http.Request
+	var nextPageURL *url.URL
+	for ok := true; ok; ok = nextPageURL != nil {
+		stats.totalPages++
+
+		if request == nil {
+			request, err = client.NewResourceRequestWithSearch(resourceType, query)
+		} else {
+			request, err = client.NewPaginatedResourceRequest(nextPageURL)
+		}
+
+		if err != nil {
+			return nil, fmt.Errorf("could not create FHIR server request: %v\n", err)
+		}
+
+		requestStart := time.Now()
+		response, err := client.Do(request)
+		if err != nil {
+			return nil, fmt.Errorf("could not request the FHIR server: %v\n", err)
+		}
+		stats.requestDurations = append(stats.requestDurations, time.Since(requestStart).Seconds())
+
+		procStart := time.Now()
```

Comment:
please use 

trace := &httptrace.ClientTrace{
		GotConn: func(_ httptrace.GotConnInfo) {
			requestStart = time.Now()
		},
		WroteRequest: func(_ httptrace.WroteRequestInfo) {
			processingStart = time.Now()
		},
		GotFirstResponseByte: func() {
			processingDuration = time.Since(processingStart)
		},
	}
	
for duration calculation, the same as in the upload command. processing is not the processing of the body, it is the processing time of the server without the transfer time